### PR TITLE
Resume unfinished GS copies

### PR DESCRIPTION
**Pull request by xbrianh**
This:
  1. Avoids lost state by adding choice step to SFN flow
  2. Provides 10 seconds of padding for thread to time out before Lambda times out

### CODE REVIEW

**Comment by ttung** [comment]
I don't think I'm qualified to review this any more.

File: dss/stepfunctions/gscopyclient/implementation.py [hunk 1]
```diff
@@ -140,6 +140,19 @@ def _sfn():
                 "Resource": copy_worker,
                 "Retry": _retry_default(),
                 "Catch": _catch_default(),
+                "Next": "TestFinished",
+            },
+            f"TestFinished": {
```
Comment (amarjandu) [inline dss/stepfunctions/gscopyclient/implementation.py]: is this supposed to be an f-string?

File: dss/stepfunctions/gscopyclient/implementation.py [hunk 2]
```diff
@@ -139,6 +139,19 @@ def _sfn():
...
+                "Type": "Choice",
+                "Choices": [{
+                    "Variable": "$.finished",
```
Comment (amarjandu) [inline dss/stepfunctions/gscopyclient/implementation.py]: please excuse this dumb question: if the variable value is not marked as "true" what mechanism saves the context of the state so it's not lost?

File: dss/stepfunctions/gscopyclient/implementation.py (hunk 1 above)
Comment (xbrianh) [reply dss/stepfunctions/gscopyclient/implementation.py]: will fix

File: dss/stepfunctions/gscopyclient/implementation.py (hunk 2 above)
Comment (xbrianh) [reply dss/stepfunctions/gscopyclient/implementation.py]: The state is saved by the AWS step function (SFN) execution engine.

The SFN calls Lambda functions on some state and saves the state returned. The [SFN language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html) also provides some flow control steps, such as the `Choice` step used here.

**Pull request merged**